src/player/player.gd

```diff
@@ -48,8 +53,14 @@ func _move(_input_dir:Vector2) -> void:
 
 	if next_tile != null:
 		print(next_tile)
+		if !has_boots and next_tile.is_in_group('chest'):
```

Comment:
_move should JUST be logic related to checking the next tile and moving the player onto another tile. The new tile should be passed into another tile to be assessed there.
I also included a bit of logic at the end to check for stairs in my branch but I shouldn't have.